notes/stv/20220616-benchmark-tests-01.txt

```diff
@@ -0,0 +1,2200 @@
+		},
+		"logs": ""
+	},
+	"Source_counts_over_the_sky.json": {
+		"result": "PASS",
+		"outputs": {
+			"valid": true
+		},
+		"time": {
+			"result": "FAST",
+			"elapsed": "19.55",
+			"expected": "22.00",
+			"percent": "-11.15",
+			"start": "2022-06-17T12:08:59.406708",
+			"finish": "2022-06-17T12:09:18.953103"
+		},
+		"logs": ""
+	},
+	"Library_Validation.json": {
+		"result": "PASS",
+		"outputs": {
+			"valid": true
+		},
+		"time": {
+			"result": "FAST",
+			"elapsed": "25.41",
+			"expected": "60.00",
+			"percent": "-57.65",
+			"start": "2022-06-17T12:09:28.962671",
+			"finish": "2022-06-17T12:09:54.374882"
+		},
+		"logs": ""
+	}
+}]
+ 
+# Not sure how to validate that this works, as the notebooks don't all start at the same time anyway for a given user.
+
+
+    
+# -----------------------------------------------------
+# Stop zeppelin and Run test, to see if we get meaningfull feedback
+# Users: 1
+# Delay_start: 0
+# Delay_notebook: 0
+# Delete Notebooks: True
+
+
+
+cat > /tmp/testprog.py << EOF
+import sys
+from aglais_benchmark import AglaisBenchmarker
+AglaisBenchmarker(
+    "/deployments/zeppelin/test/config/quick.json",
+    "/tmp/testusers.json",
+    "/tmp/",
+    "http://128.232.227.238:8080/",
+    False
+    ).run(
+        concurrent=False,
+        users=1,
+        delay_start=0,
+        delay_notebook=10,
+        delete=True
+        )
+
+EOF
+
+python3 /tmp/testprog.py | tee /tmp/test-results.txt
+
+# Result:
+
+Exception encountered while trying to create a notebook: /tmp/7JN2WBGP5I.json for user in config: /tmp/user.yml
+HTTPConnectionPool(host='128.232.227.238', port=8080): Max retries exceeded with url: /api/login (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fd0d25f4880>: Failed to establish a new connection: [Errno 111] Connection refused'))
+
+Exception encountered while trying to create a notebook: /tmp/C1UJTCPMXB.json for user in config: /tmp/user.yml
+HTTPConnectionPool(host='128.232.227.238', port=8080): Max retries exceeded with url: /api/login (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f1cc1db0880>: Failed to establish a new connection: [Errno 111] Connection refused'))
+
+Exception encountered while trying to create a notebook: /tmp/MJ26A8XLYP.json for user in config: /tmp/user.yml
+HTTPConnectionPool(host='128.232.227.238', port=8080): Max retries exceeded with url: /api/login (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fd20c7f0880>: Failed to establish a new connection: [Errno 111] Connection refused'))
+
+Exception encountered while trying to create a notebook: /tmp/I0EZRUTY3Y.json for user in config: /tmp/user.yml
+HTTPConnectionPool(host='128.232.227.238', port=8080): Max retries exceeded with url: /api/login (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fa7581f0880>: Failed to establish a new connection: [Errno 111] Connection refused'))
+
```

Comment:
These need to be part of the JSON output, linked to the results for notebook that failed. So that the output from a whole test run can be treated as a JSON fragment that can be embedded in a larger document.

For example :
```
    {
    "GaiaDMPSetup": {
        "result": "FAIL",
        "messages" [
            "Exception encountered while trying to create a notebook: /tmp/7JN2WBGP5I.json for user in config: /tmp/user.yml",
            "HTTPConnectionPool(host='128.232.227.238', port=8080): Max retries exceeded with url: /api/login (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fd0d25f4880>: Failed to establish a new connection: [Errno 111] Connection refused'))"
            ]
        "outputs": {
            "valid": true
            },
        "time": {
            "result": "FAST",
            "elapsed": "2.28",
            "expected": "45.00",
            "percent": "-94.93",
            "start": "2022-06-17T12:15:05.706600",
            "finish": "2022-06-17T12:15:07.987354"
            },
        "logs": ""
        },
    "Mean_proper_motions_over_the_sky": {
        "result": "FAIL",
        "messages" [
            "Exception encountered while trying to create a notebook: /tmp/C1UJTCPMXB.json for user in config: /tmp/user.yml",
            "HTTPConnectionPool(host='128.232.227.238', port=8080): Max retries exceeded with url: /api/login (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f1cc1db0880>: Failed to establish a new connection: [Errno 111] Connection refused'))"
            ]
        "outputs": {
            "valid": true
            },
        "time": {
            "result": "FAST",
            "elapsed": "1.29",
            "expected": "55.00",
            "percent": "-97.66",
            "start": "2022-06-17T12:15:17.994824",
            "finish": "2022-06-17T12:15:19.283176"
            },
        "logs": ""
        },
    ....
    ....
    }
```

This whole block can then be embedded in the results of a larger test that runs the benchmark multiple times.

Reply:
Ok thanks for the suggestion, I've gone ahead and made the changes to the benchmarker to store the errors in the output json.